neo4j-ogm-docs/src/main/asciidoc/reference/nativetypes.adoc

```diff
@@ -0,0 +1,183 @@
+If you are sure that you only deal with one or the other, you just use an explicit mapping to either `java.time.Duration` or `java.time.Period`.
+
++**+ There is no generic Java type representing a spatial point.
+As OGM supports different ways of connecting to Neo4j it cannot expose either the Java drivers or the internal representation of a point, so it provides a point of it's own.
+Please read the next section to learn which concrete classes Neo4j-OGM offers for a point.
+======
+
+[[reference:native-property-types:spatial-types]]
+== Mapping of Neo4j spatial types
+
+Neo4j supports four slightly different property types for spatial points, see https://neo4j.com/docs/cypher-manual/current/syntax/spatial/[Spatial values].
+All variations of the `Point` type are backed by an index and therefore perform very well in queries.
+The main difference between them is the coordinate reference system.
+A point can either be stored in a geographic coordinate system with longitude and latitude or in a cartesian system with x and y.
+If you add the third dimension, you add altitude or a z-axis.
+
+Geographic coordinate systems are based on a spheroidal surface.
+Coordinates are projections of points on that surface to a flat plane.
+Attributes of type `Point` in Neo4j having geographic coordinates return `longitude` and `latitude` with a fixed reference system of WGS-84 (SRID 4326, the same one that many online mapservers use).
+3-dimensional geographic coordinates have a reference system of WGS-84-3D with the SRID 4979.
+
+Cartesian coordinate systems deal with locations in an euclidean space and are not projected.
+Attributes of type `Point` in Neo4j having cartesian coordinates return `x` and `y`, their SRID is 7203 respectively 9157.
+
+The important take-aways for modelling your domain is the fact that points with different coordinate systems are not comparable without undergoing transformation.
+The same attribute of a node should always be using the same coordinate system than all the other nodes with the same label.
+Otherwise the `distance` and all other functions dealing with multiple `Points` will return literal `null`.
+
+To make modelling a domain less error prone, Neo4-OGM provides four distinct types that you can use in your Neo4j entities:
+
+======
+[options="header", cols="m,"]
+|===
+| Neo4j-OGM type | Neo4j point type
+| `GeographicPoint2d` | A point with `longitude` and `latitude` in a geographic reference system of WGS-84
+| `GeographicPoint3d` | A point with `longitude`, `latitude` and `altitude` in a geographic reference system of WGS-84-3D
+| `Cartesian2d` | A point with `x` and `y` in euclidean space
+| `Cartesian3d` | A point with `x`, `y` and `z` in euclidean space
+|===
+
+Use-cases for the geographic points are all kinds of stuff you usually find on a map.
+Cartesian points are usefull for indoor navigation etc.
```

Comment:
Cartesian could be for any 2D and 3D modeling, not just in the real world.

Reply:
Units of geographic are degrees, while units of cartesian are undefined and can be any units (metres, feet, microns, foobars).